awd/article.py

```diff
@@ -18,57 +19,101 @@ class Article:
+    ) -> None:
         """Initialize article instance.
 
         Args:
             doi: A digital object identifer (doi) for an article.
             metadata_url: The URL for retrieving metadata records.
-            content_url: The URL for retrieving article content
+            content_url: The URL for retrieving article content.
+            doi_table: The DOI table as a PynamoDB object.
+            doi_table_items: A list of the DOI table items as PynamoDB objects.
         """
         self.doi: str = doi
         self.metadata_url: str = metadata_url
         self.content_url: str = content_url
-        self.database_item = None
+        self.doi_table: DoiProcessAttempt = doi_table
+        self.doi_table_items: list[DoiProcessAttempt] = doi_table_items
         self.crossref_metadata: dict[str, Any]
         self.dspace_metadata: dict[str, Any]
         self.article_content: bytes
 
     def process(self) -> None:
+        """Run the complete article processing workflow."""
+        self.add_item_to_doi_table()
         self.get_and_validate_crossref_metadata()
         self.create_and_validate_dspace_metadata()
         self.get_and_validate_wiley_article_content()
 
-    def exists_in_database(self, database_items: list[dict[str, Any]]) -> bool:
-        """Validate that a DOI is NOT in the database and needs to be added.
-
-        Args:
-            database_items: A list of database items that may or may not contain the
-            specified DOI.
-        """
+    def exists_in_doi_table(self) -> bool:
+        """Validate that a DOI is NOT in the DOI table and needs to be added."""
         exists = False
-        if not any(doi_item["doi"] == self.doi for doi_item in database_items):
+        if any(doi_item.doi == self.doi for doi_item in self.doi_table_items):
             exists = True
-            logger.debug("%s added to database", self.doi)
+            logger.debug("%s added to DOI table", self.doi)
         return exists
 
-    def has_retry_status(self, database_items: list[dict[str, Any]]) -> bool:
-        """Validate that a DOI should be retried based on its status in the database.
-
-        Args:
-            database_items: A list of database items containing the specified DOI, whose
-            status must be evaluated for whether the application should attempt to process
-            it again.
-        """
-        retry_status = False
+    def has_unprocessed_status(self) -> bool:
+        """Validate that a DOI has unprocessed status in the DOI table."""
+        unprocessed_status = False
         if any(
-            d
-            for d in database_items
-            if d["doi"] == self.doi and d["status"] == str(Status.UNPROCESSED.value)
+            doi_item
+            for doi_item in self.doi_table_items
+            if doi_item.doi == self.doi and doi_item.status == Status.UNPROCESSED.value
         ):
-            retry_status = True
+            unprocessed_status = True
```

Comment:
I wonder if you could lean into the ORM nature of pynamodb a bit more here.

I think you could do something along the lines of:
```
# retrieve single row matching on hash key
doi_process_atempt = DoiProcessAttempt.get(self.doi)

# if not exists, return False
if doi_process_attempt is None:
    return False

# return bool if has unprocessed status
return doi_process_attempt.status == Status.UNPROCESSED.value    
```

In fact, this _itself_ might make a nice method on the `DoiProcessAttempt` model.  Just a thought.